lib/src/json_schema/validator.dart

```diff
@@ -720,28 +720,32 @@ class Validator {
       instance = Instance(instance);
     }
 
+    if (schema.unevaluatedItems != null) {
+      _pushEvaluatedItemsContext();
+    }
+    if (schema.unevaluatedProperties != null) {
+      _pushEvaluatedPropertiesContext();
+    }
+
     /// If the [JsonSchema] being validated is a ref, pull the ref
     /// from the [refMap] instead.
-    while (schema.ref != null || schema.recursiveRef != null) {
+    if (schema.ref != null || schema.recursiveRef != null) {
       var nextSchema = schema.resolvePath(schema.ref ?? schema.recursiveRef);
       if (schema.recursiveRef != null && nextSchema.recursiveAnchor == true) {
```

Comment:
Is it valid to have a `$ref` alongside a `$recursiveRef`?

Reply:
Good Question. I'll figure that out and update this implementation.

Reply:
While seemingly insane, it does look like it is NOT invalid to have a `$ref` along side a `$recursiveRef`.